src/rust/engine/process_execution/src/remote.rs

```diff
@@ -38,36 +49,30 @@ impl CommandRunner {
   pub fn run_command_remote(
     &self,
     req: ExecuteProcessRequest,
-  ) -> Result<ExecuteProcessResult, String> {
-    let execute_request = make_execute_request(&req)?;
-
-    let initial_result = map_grpc_result(self.execution_client.execute(&execute_request))?;
-
-    match extract_execute_response(&initial_result)? {
-      Some(value) => {
-        return Ok(value);
-      }
-      None => {}
-    }
-
-    let mut operation_request = bazel_protos::operations::GetOperationRequest::new();
-    operation_request.set_name(initial_result.get_name().to_string());
-    loop {
-      // TODO: Use some better looping-frequency strategy than a tight-loop.
-      let operation_result =
-        map_grpc_result(self.operations_client.get_operation(&operation_request))?;
-
-      let result = extract_execute_response(&operation_result)?;
-
-      match result {
-        Some(value) => {
-          break Ok(value);
-        }
-        None => {
-          continue;
-        }
-      }
-    }
+  ) -> BoxFuture<ExecuteProcessResult, String> {
+    let execution_client = self.execution_client.clone();
+    let operations_client = self.operations_client.clone();
+    future::done(make_execute_request(&req))
+        .and_then(move |execute_request| {
+          map_grpc_result(execution_client.execute(&execute_request))
+        })
+        // TODO: Use some better looping-frequency strategy than a tight-loop.
+        // TODO: Add a timeout of some kind.
```

Comment:
Is this ticket worthy at this point? Should be a thing you could farm out.

Reply:
Yeah, I was going to file some tickets for these after this PR landed, so that I could link to code in the tickets. Might as well file them now, I guess :) Filed https://github.com/pantsbuild/pants/issues/5503 and https://github.com/pantsbuild/pants/issues/5504 and linked to them.